File: app/src/main/java/me/tagavari/airmessage/redux/ReduxEmitterNetwork.java
package me.tagavari.airmessage.redux;

import io.reactivex.rxjava3.subjects.BehaviorSubject;
import io.reactivex.rxjava3.subjects.PublishSubject;

public final class ReduxEmitterNetwork {
	private static final BehaviorSubject<ReduxEventConnection> connectionStateSubject = BehaviorSubject.create();
	private static final PublishSubject<Boolean> connectionConfigurationSubject = PublishSubject.create();
	private static final PublishSubject<ReduxEventMessaging> messageUpdateSubject = PublishSubject.create();
	private static final BehaviorSubject<ReduxEventMassRetrieval> massRetrievalUpdateSubject = BehaviorSubject.create();
	private static final BehaviorSubject<ReduxEventTextImport> textImportUpdateSubject = BehaviorSubject.create();
	
	public static BehaviorSubject<ReduxEventConnection> getConnectionStateSubject() {
		return connectionStateSubject;
	}
	
	public static PublishSubject<Boolean> getConnectionConfigurationSubject() {
		return connectionConfigurationSubject;
	}
	
	public static PublishSubject<ReduxEventMessaging> getMessageUpdateSubject() {
		return messageUpdateSubject;
	}
	
	public static BehaviorSubject<ReduxEventMassRetrieval> getMassRetrievalUpdateSubject() {
		return massRetrievalUpdateSubject;
	}
	
	public static BehaviorSubject<ReduxEventTextImport> getTextImportUpdateSubject() {
		return textImportUpdateSubject;
	}
}

File: app/src/main/java/me/tagavari/airmessage/redux/ReduxEmitterNetwork.kt
package me.tagavari.airmessage.redux

import io.reactivex.rxjava3.subjects.BehaviorSubject
import io.reactivex.rxjava3.subjects.PublishSubject
import me.tagavari.airmessage.util.ServerUpdateData
import java.util.*

object ReduxEmitterNetwork {
	@JvmStatic
	val connectionStateSubject: BehaviorSubject<ReduxEventConnection> = BehaviorSubject.create()
	
	@JvmStatic
	val connectionConfigurationSubject: PublishSubject<Boolean> = PublishSubject.create()
	
	@JvmStatic
	val messageUpdateSubject: PublishSubject<ReduxEventMessaging> = PublishSubject.create()
	
	@JvmStatic
	val massRetrievalUpdateSubject: BehaviorSubject<ReduxEventMassRetrieval> = BehaviorSubject.create()
	
	@JvmStatic
	val textImportUpdateSubject: PublishSubject<ReduxEventTextImport> = PublishSubject.create()

	@JvmStatic
	val remoteUpdateSubject: BehaviorSubject<Optional<ServerUpdateData>> = BehaviorSubject.create()

	@JvmStatic
	val remoteUpdateProgressSubject: PublishSubject<ReduxEventRemoteUpdate> = PublishSubject.create()

	@JvmStatic
	val serverFaceTimeSupportSubject: BehaviorSubject<Boolean> = BehaviorSubject.create()
	
	@JvmStatic
	val faceTimeUpdateSubject: PublishSubject<ReduxEventFaceTime> = PublishSubject.create()
	
	@JvmStatic
	val faceTimeIncomingCallerSubject: BehaviorSubject<Optional<String>> = BehaviorSubject.create()
}